cobra/core/Model.py

```diff
@@ -230,9 +308,85 @@ def add_reactions(self, reaction_list):
 
         self.reactions += reaction_list
 
+        # from cameo ...
+        self._populate_solver(reaction_list)
+
+    def _populate_solver(self, reaction_list, metabolite_list=None):
+        """Populate attached solver with constraints and variables that
+        model the provided reactions.
+        """
+        constraint_terms = AutoVivification()
+        if metabolite_list is not None:
```

Comment:
Usually that's true, but in the case where you have a metabolite that is not in any of the reactions in reaction_list you need to add the empty constraint separately. (So my previous answer was actually not correct. I was referring to the Constraint construction in line 363, this one is only there to make sure all metabolites get a constraint, even if they don't have any reactions)